dashboard/grid_builder.py

```diff
@@ -70,13 +72,19 @@ def _is_py_version_incompatible(self, result):
 
     def has_issues(self, p: package.Package) -> bool:
         """Returns true if the given package has any compatibility issues."""
+        # Get self result
         for package_2 in self._package_to_results.keys():
-            result = self.get_result(p, package_2)
+            results = self.get_result(p, package_2)
             # Don't report the package as having issues if it is purely the
             # result of a self-incompatibility of another package.
-            if result['status'] != compatibility_store.Status.SUCCESS.name and (
-                    not result['self'] or p == package_2):
-                return True
+            pure_self_conflict = False
```

Comment:
What does "pure_self_conflict" mean?

Reply:
It means the pairwise check failure is because one of the packages is not conflict with itself. And it enables us to differentiate the two types of pairwise failures in the dashboard as described below:
e.g google-cloud-dataflow and tensorflow is failing with status color purple(self conflict), while google-cloud-dataflow and google-cloud-language is failing with status color orange(pairwise conflict).

Reply:
Done.